tests/unit/test_allocator.cpp

```diff
@@ -150,19 +204,21 @@ TEST_F(AllocatorTest, test_hnsw) {
 
     float vec[128] = {};
     HNSWIndex *hnswIndex = new (allocator) HNSWIndex(&params, allocator);
-    ASSERT_GE(allocator->getAllocationSize(), sizeof(VecSimAllocator) + sizeof(HNSWIndex));
+    expectedAllocationSize +=
+        sizeof(BruteForceIndex) + sizeof(InnerProductSpace) + 2 * vecsimAllocationOverhead;
+    ASSERT_GE(allocator->getAllocationSize(), expectedAllocationSize);
```

Comment:
Can we use `ASSERT_EQ` here, since the capacity is 0?

Reply:
nope, as there are STLs being build as well as memory allocations happens in HNSW Ctor